### ISSUE TITLE
Dataclass fields that use `converter` don't have expected semantic tokens

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#6525]
@debonte the ` __converterDescriptor_x` regression was addressed in pyright 1.1.382.

But there remains one more issue. If an attribute uses a converter then it is displayed in white, not light blue. Here is a simple example with longer names so you can see the effect clearly.
```python
from attrs import define, field


@define
class A:
    normal_attribute: int = field()
    converted_attribute: int = field(converter=int)


a = A(1, 2)
print(a.normal_attribute)
print(a.converted_attribute)
```
![image](https://github.com/user-attachments/assets/8771801b-6d63-48e2-a973-3d8362c11d60)

According to Eric (https://github.com/microsoft/pyright/issues/8881#issuecomment-2327140623)

> The highlight color is provided by pylance's "semantic token provider". This functionality isn't part of pyright, so the pylance team will need to address that aspect separately if it's still an issue.

_Originally posted by @my1e5 in https://github.com/microsoft/pylance-release/issues/5391#issuecomment-2393888089_

**Comment by debonte** [comment]
Additionally, go to def fails on `a.converted_attribute`. Both issues are happening because we're not creating a declaration for `converted_attribute` when the descriptor type is synthesized. I'll send a PR to Pyright.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2024.10.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202410100-10-October-2024-PreRelease)
